src/backend/distributed/commands/table.c

```diff
@@ -1886,6 +1917,53 @@ get_attrdef_oid(Oid relationId, AttrNumber attnum)
 }
 
 
+/*
+ * AppendAlterTableWithNextvalDefaultStmt appends a string representing either:
+ * ALTER TABLE ADD COLUMN .. DEFAULT nextval(), if TypeName isn't empty, or
+ * ALTER TABLE ALTER COLUMN .. SET DEFAULT nextval()
+ * If sequence type is not bigint, we use worker_nextval() instead of nextval().
+ */
+static void
+AppendAlterTableWithNextvalDefaultStmt(StringInfo buf, Oid sequenceOid, Oid relationId,
+									   char *colname, TypeName *typeName)
+{
+	char *qualifiedSequenceName = generate_qualified_relation_name(sequenceOid);
+	char *qualifiedRelationName = generate_qualified_relation_name(relationId);
+
+	char *nextval_expr = "nextval";
+	bool use_worker_nextval = (pg_get_sequencedef(sequenceOid)->seqtypid != INT8OID);
+	if (use_worker_nextval)
+	{
+		/*
+		 * We use worker_nextval for int and smallint types.
+		 * Check issue #5126 and PR #5254 for details.
+		 * https://github.com/citusdata/citus/issues/5126
+		 */
+		nextval_expr = "worker_nextval";
+	}
+
+	if (typeName == NULL)
+	{
+		appendStringInfo(buf, "ALTER TABLE %s ALTER COLUMN %s "
+							  "SET DEFAULT %s(%s::regclass)",
+						 qualifiedRelationName, colname, nextval_expr,
```

Comment:
maybe instead of nextval_expr, we could use `quote_qualified_identifier("pg_catalog", nextvalFunctionName)`